web_access_rule_buttons/models.py

```diff
@@ -28,4 +28,11 @@ def check_access_rule_all(self, operations=None):
     return result
 
 
-models.BaseModel.check_access_rule_all = check_access_rule_all
+# Could be any model, we just use a core model to have a 'register_hook'
+class IrModel(models.Model):
+    _inherit = 'ir.model'
+
+    def _register_hook(self, cr):
+        # Add method check_access_rule_all for all models
+        models.BaseModel.check_access_rule_all = check_access_rule_all
```

Comment:
That's right. But I don't think there is any solution to that. Another issue is that if you uninstall an addon, the code changed by `_register_hook` stays. This is a huge weakness of the way Odoo loads addons. This is by far not the only addon to suffer from that. Hopefully in v10, we'll be able to simply inherits the model 'base': https://github.com/odoo/odoo/pull/10288/files#diff-678bd201dfe1a0c69add630cf73e76baR58

But again, the worst case is that this method is added on the models, but will never be called because the js code that calls it is not loaded so that's not dramatic.